### DISCUSSION TITLE
테스트 구조에 대해서

### DISCUSSION THREAD

**Discussion by shkisme** [BDD-CLUB/01-doo-re-back#33] [논의 사항]
저희가 테스트 하기로 한 범위와 방식은 다음과 같습니다.
- Controller api 테스트 진행 (@SpringBootTest)
- Rest Docs 테스트 진행 (Mocking하여 진행)
- Service 테스트 진행 (Mocking 하지 않고, 필요한 경우에만 Mocking. @SpringBootTest)
- Repository 테스트 진행 (직접 작성한 쿼리만. @DataJpaTest)
- Pojo 테스트

따라서 제가 구성해 본 테스트 구조는 다음과 같습니다. (예제로는 현재 개발 중인 팀 도메인을 들었습니다.)
<img width="1254" alt="image" src="https://github.com/BDD-CLUB/01-doo-re-back/assets/92802207/829f3a0e-c8c1-44ac-b7b1-72049e0f391e">

https://github.com/BDD-CLUB/01-doo-re-back/pull/32 여기서 코드를 직접 보실 수도 있습니다.

최상위 클래스 네이밍도 우선 제가 임의로 정해 본 것인데 이것도 논의해서 결정되면 좋겠습니다.

+) 폴더 구조 
<img width="480" alt="image" src="https://github.com/BDD-CLUB/01-doo-re-back/assets/92802207/e702e979-3b02-4440-8ad2-68abb893809b">

**Comment by lcqff** [comment]
좋습니다~! 이게 최선인거 같아보입니다!

**Comment by 02ggang9** [comment]
구조랑 이름 둘 다 마음에 듭니다~

**Comment by amaran-th** [comment]
Repository 테스트는 integration 테스트보다는 slice 테스트라고 하는 편이 좀 더 적절해 보입니다!
그래서 패키지 이름도 integration보다는 다른 이름으로 지으면 좋을 것 같아요~ 가령 helper라든지요!

**Reply by shkisme** [reply]
Spring에서 Repository 레이어만 잘라서 테스트한다는 의미에서 Slice라고 명명하는 편인가요? 처음 알았네요..
저는 Repository 레이어를 통합하여 테스트한다는 의미에서 RepositoryIntegrationTest라 명명했었습니다!

혹시 패키지 이름은 왜 helper가 되어야 할까요? ~helper는 유틸리티의 느낌이 강해서 해당 폴더에 최상위 클래스를 두는게 맞는 지 잘 모르겠습니다! 물론 저도 개발할 때 최상위 클래스에 유틸리티용 메서드를 위치시켜 놓긴 해서 Helper로 보는 게 맞나? 싶긴 했었습니다! 의견 궁금합니다!!

**Comment by JJimini** [comment]
확인했습니다. 그대로 진행하시면 좋을 것 같아요

**Comment by shkisme** [comment]
📌 회의 결과 📌

- Repository 빈에 관한 테스트만 진행하므로, `RepositorySliceTest`로 네이밍 확정
- 폴더 구조 네이밍은 명확하게 떠오리는 이름이 없어서 임시로 helper로 결정. 추후 더 적합한 네이밍이 나오면 해당 이름으로 변경하도록 결정.
